lib/cylc/config.py

```diff
@@ -1522,78 +1520,68 @@ def generate_triggers(self, lexpression, left_nodes, right, seq,
         except SyntaxError:
             raise SuiteConfigError('Error in expression "%s"' % lexpression)
 
-        def recursive_replace(lst, obj, replacement):
-            """Replace occurances of an object in a list with a standin.
-
-            Works recursively to replace instances in nested lists.
-
-            """
-            ret = []
-            for item in lst:
-                if type(item) is list:
-                    ret.append(recursive_replace(item, obj, replacement))
-                elif item == obj:
-                    ret.append(replacement)
-                else:
-                    ret.append(item)
-            return ret
-
-        trigers = set([])
+        triggers = {}
         for left in left_nodes:
             # (GraphNodeError checked above)
-            lnode = graphnode(left, base_interval=base_interval)
-            ltaskdef = self.taskdefs[lnode.name]
+            name, offset_is_from_icp, offset_is_irregular, offset, output = (
+                GraphNodeParser.get_inst().parse(left))
+            ltaskdef = self.taskdefs[name]
 
             # Determine intercycle offsets.
             abs_cycle_point = None
             cycle_point_offset = None
-            if lnode.offset_is_from_ict:
-                first_point = get_point_relative(
-                    lnode.offset_string, self.initial_point)
+            if offset_is_from_icp:
+                first_point = get_point_relative(offset, self.initial_point)
                 last_point = seq.get_stop_point()
                 abs_cycle_point = first_point
                 if last_point is None:
                     # This dependency persists for the whole suite run.
-                    ltaskdef.intercycle_offsets.add(
-                        (None, seq))
+                    ltaskdef.intercycle_offsets.add((None, seq))
                 else:
                     ltaskdef.intercycle_offsets.add(
-                        (str(-(last_point - first_point)), seq))
-            elif lnode.is_absolute:
-                abs_cycle_point = lnode.offset_string
-            elif lnode.intercycle:
-                if lnode.offset_is_irregular:
-                    offset_tuple = (lnode.offset_string, seq)
+                        (str(first_point - last_point), seq))
+            elif offset_is_from_icp:
```

Comment:
In the old logic, `is_absolute` is always set to the same value as `offset_is_from_ict`. In the new logic, we have unified the variables as `offset_is_from_icp`.